### PR TITLE
Add testAuthenticationGet

### PR DESCRIPTION
**Pull request by djmitche**
A new method for testing bewits, which only work for GET requests.  Plus tests.

### CODE REVIEW

**Comment by djmitche** [comment]
I just updated this to take a parameter instead, since to test named temporary credentials, I will need to have the clientId associated with `auth:create-client:<clientId>`.

File: auth/v1.js [hunk 1]
```diff
@@ -816,6 +816,55 @@ api.declare({
...
+  stability:  'experimental',
+  title:      "Test Authentication (GET)",
+  description: [
+    "Utility method similar to `testAuthenticate`, but with the GET method,",
+    "so it can be used with signed URLs (bewits).",
+    "",
+    "Rather than using real credentials, this endpoint accepts requests with",
+    "clientId `tester` and accessToken `no-secret`. That client's scopes are",
+    "`['test-authenticate-get', clientScope]`.  The call fails if the",
+    "`test-authenticate-get` scope is not avaialble (e.g,. via authenticateScopes)",
```
Comment (jonasfj) [inline auth/v1.js]: If we really want to test the clients... we should make this use querystring for the two lists of scopes we had before...
Just say the querystring values have to JSON encoded lists of scopes. That should be easy to play with...

I suspect most of our client libraries either don't support querystring options yet, or don't have serious tests for them... And certainly not in combination with bewits...
We could say that if no query string is provided, then these hardcoded defaults are assumed.

That way the end-point can be used for testing both cases...


----
Okay, I guess in that way, we can just extend this end-point later... No need to do it now...

Note: query string parameters are always optional..

**Comment by jonasfj** [comment]
`r+`

**Comment by djmitche** [comment]
I think this is pretty ugly, embedding a (single) scope as parameter, but it works and avoids issues with clients that don't yet support queries.

**Comment by djmitche** [comment]
Ideally, we'd have `?clientScope=..&clientScope=..` etc. Same for `requiredScope`.

One option is to just hard-code the clientScope and requiredScope here, and later allow those to be overridden by query args (since not all clients support query args yet).  That would allow easy expansion later without breaking existing users of the endpoint.

**Comment by jonasfj** [comment]
`r+`

**Pull request merged**